src/DSharp.Compiler/CodeModel/Names/AtomicNameNode.cs

```diff
@@ -23,5 +24,11 @@ protected AtomicNameNode(ParseNodeType nodeType, IdentifierToken identifier)
         public IdentifierToken Identifier { get; }
 
         protected sealed override ParseNodeList List => new ParseNodeList(this);
+
+        public bool Equals(AtomicNameNode other)
+        {
+            return other.Identifier.Type == Identifier.Type
```

Comment:
I know this is a stupid thing but do we need to null check other?

Reply:
We should yes